ADDING_A_PLUGIN.md

```diff
@@ -0,0 +1,347 @@
+# Developing a TensorBoard plugin
+
+<!-- This document was last reviewed on July 21, 2017. It should be
+reviewed occasionally to make sure it stays up-to-date. -->
+
+## Overview
+
+You know (and, we hope, love!) TensorBoard's core features like the scalar dashboard, the graph explorer, and the embedding projector. However, in every TensorBoard user's life, there comes a time when they want some cool new visualization that just…doesn't exist yet. That's what this plugin system is for.
```

Comment:
One reason for using *they* here is that *he or she* excludes non-binary genders.

(Given that the OED lists this as def. 2 of *they*, noting "[e]specially in relation to a noun phrase involving one of the indefinite determiners or pronouns *any*, *each*, *every*, *no*, *some*, *anybody*, *anyone*, etc.", and with references dating to 1375, I'm fine with its use.)